admin/actions.py

```diff
@@ -70,16 +75,74 @@ def index(self):
                         "Submission ID %s reingested by %s", form.submission_id.data, g.user.email
                     )
                     flash(f"Successfully re-ingested submission {submission.submission_id}", "success")
+
                 else:
                     flash(f"Issues re-ingesting submission {submission.submission_id}: {response_data}", "error")
 
-                return redirect(url_for("reingest.index"))
+                return redirect(url_for("reingest_s3.index"))
 
             flash_errors(form, "%(error)s")
 
         return self.render("admin/reingest.html", form=form)
 
 
+class ReingestFileAdminView(BaseAdminView):
+    @expose("/", methods=["GET", "POST"])
+    def index(self):
+        form = ReingestFromFileAdminForm(CombinedMultiDict((request.files, request.form)))
+
+        if form.is_submitted():
+            if form.validate():
+                submission_id = form.submission_id.data
+
+                try:
+                    submission = Submission.query.filter_by(submission_id=submission_id).one()
+                except NoResultFound as error:
+                    raise error
+
+                if submission:
+                    match submission.programme_junction.programme_ref.fund.fund_code:
+                        case "PF":
+                            fund_name = "Pathfinders"
+                        case "TD" | "HS":
+                            fund_name = "Towns Fund"
+                        case fund_code:
+                            raise ValueError(f"Unknown fund: {fund_code}")
+
+                    reporting_round = submission.reporting_round.round_number
+                    account_id, user_email = submission.submitting_account_id, submission.submitting_user_email
```

Comment:
Where are these values coming from? Are they the logged in user? Ir whoever did the original submission?

Reply:
Yes, whoever did the original submission.

This is copied from the original CLI reingest_file, so maintaining that behaviour. If we want to change that now is, I suppose, up for discussion :)